test/octree/test_octree.cpp

```diff
@@ -1367,83 +1379,85 @@ TEST (PCL, Octree_Pointcloud_Neighbours_Within_Radius_Search)
 
   // instantiate point clouds
   PointCloud<PointXYZ>::Ptr cloudIn (new PointCloud<PointXYZ> ());
-  PointCloud<PointXYZ>::Ptr cloudOut (new PointCloud<PointXYZ> ());
 
   srand (static_cast<unsigned int> (time (nullptr)));
 
-  for (unsigned int test_id = 0; test_id < test_runs; test_id++)
-  {
-    // define a random search point
-    PointXYZ searchPoint (static_cast<float> (10.0 * rand () / RAND_MAX),
-                          static_cast<float> (10.0 * rand () / RAND_MAX),
-                          static_cast<float> (10.0 * rand () / RAND_MAX));
-
-    cloudIn->width = 1000;
-    cloudIn->height = 1;
-    cloudIn->points.resize (cloudIn->width * cloudIn->height);
-
-    // generate point cloud data
-    for (std::size_t i = 0; i < 1000; i++)
+  for (const std::size_t maxObjsPerLeaf: {0, 5}) {
+    for (unsigned int test_id = 0; test_id < test_runs; test_id++)
     {
-      (*cloudIn)[i] = PointXYZ (static_cast<float> (10.0 * rand () / RAND_MAX),
-                                     static_cast<float> (10.0 * rand () / RAND_MAX),
-                                     static_cast<float> (5.0  * rand () / RAND_MAX));
-    }
+      // define a random search point
+      PointXYZ searchPoint (static_cast<float> (10.0 * rand () / RAND_MAX),
+                            static_cast<float> (10.0 * rand () / RAND_MAX),
+                            static_cast<float> (10.0 * rand () / RAND_MAX));
```

Comment:
Reason why Z value should be further out than the generated test cloud? (10.0 vs 5.0)

Reply:
Yes, probably.